src/controllers/oauthContoller.ts

```diff
@@ -61,10 +61,19 @@ export const handleAuthorize = async (req: Request, res: Response) => {
     // nonce will be part of id token (jwt) claims
     let isUpdated;
     if (req.body.code_challenge) {
-      isUpdated = await saveStateCodeChallengeAndNonce(
-        code.authorizationCode,
-        req.body
-      ); // saving state, code_challenge and nonce
+      if (
```

Comment:
check this first, before sending mail and db queries, better create a new middleware and add it first in the request-response chain